pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -296,21 +299,23 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 
 // getPeerConfig returns the CiliumBGPPeerConfigSpec for the given peerConfigName. If config does not exist, nil is returned.
 func (r *NeighborReconciler) getPeerConfig(peerConfigName string) (*v2alpha1.CiliumBGPPeerConfigSpec, error) {
+	if peerConfigName == "" {
+		// if peer config is not specified, return default config
+		conf := &v2alpha1.CiliumBGPPeerConfigSpec{}
+		conf.SetDefaults()
+		return conf, nil
+	}
+
 	config, exists, err := r.PeerConfig.GetByKey(resource.Key{Name: peerConfigName})
 	if err != nil {
 		return nil, err
 	}
-
-	var conf *v2alpha1.CiliumBGPPeerConfigSpec
 	if !exists {
-		// if config does not exist, return default config
-		conf = &v2alpha1.CiliumBGPPeerConfigSpec{}
-	} else {
-		conf = &config.Spec
+		return nil, nil // peer config specified, but not found - return nil
```

Comment:
Should an error be returned instead? Passing nil config might cause issues in underlying layers which rely on it being set.

Reply:
Ah gobgp layer will return "nil peer config in NeighborRequest". It might be better to return error from here since we have more context why it is nil.

Reply:
nil would never be passed to lower levels as it was checked & skipped right after calling getPeerConfig ... but I added `exists` return value to `getPeerConfig`, so that we do not have to rely on nil check in the logic.